sky/clouds/gcp.py

```diff
@@ -792,6 +792,54 @@ def _get_disk_type(cls, disk_tier: Optional[str]) -> str:
     def _label_filter_str(cls, tag_filters: Dict[str, str]) -> str:
         return ' '.join(f'labels.{k}={v}' for k, v in tag_filters.items())
 
+    @classmethod
+    def check_quota_available(cls, resources: 'resources.Resources') -> bool:
+        """Check if GCP quota is available based on `resources`.
+
+        GCP-specific implementation of check_quota_available. The function works by
+        matching the `accelerator` to the a corresponding GCP keyword, and then using
+        the GCP CLI commands to query for the specific quota (the `accelerator` as
+        defined by `resources`).
+
+        Returns:
+            False if the quota is found to be zero, and True otherwise.
+        Raises:
+            CalledProcessError: error with the GCP CLI command.
+        """
+
+        if not resources.accelerators:
+            return True
```

Comment:
Can we add quick comment here mentioning why we take this exit if accelerators are not specified?
```suggestion
            # TODO(hriday): We currently only support checking quotas for GPUs. 
            #   For CPU-only instances, we need to try provisioning to check quotas.
            return True
```